packages/web/app/components/Medication/MedicationImportModal.jsx

```diff
@@ -131,8 +132,30 @@ export const MedicationImportModal = ({ encounter, open, onClose, onSaved }) =>
   const [prescriberId, setPrescriberId] = useState(null);
   const queryClient = useQueryClient();
 
-  const { data, isLoading, error } = usePatientOngoingPrescriptionsQuery(encounter.patientId);
-  const medications = useMemo(() => data?.data.filter(p => !p.discontinued) || [], [data]);
+  const { data: encounterPrescriptionsData } = useEncounterMedicationQuery(encounter.id);
+  const {
+    data: patientOngoingPrescriptionsData,
+    isLoading,
+    error,
+  } = usePatientOngoingPrescriptionsQuery(encounter.patientId);
+
+  const medications = useMemo(() => {
+    const encounterPrescriptions = encounterPrescriptionsData?.data || [];
+    const patientOngoingPrescriptions = patientOngoingPrescriptionsData?.data || [];
+    return patientOngoingPrescriptions
+      .filter(p => !p.discontinued)
+      .filter(
+        p =>
+          !encounterPrescriptions.some(
+            ep =>
+              ep.medicationId === p.medicationId &&
+              ep.doseAmount === p.doseAmount &&
+              ep.units === p.units &&
+              ep.route === p.route &&
+              ep.frequency === p.frequency,
+          ),
```

Comment:
This inner loop could be avoided by creating a `Set` with a hash of the important details